client/src/app/components/Autocomplete.tsx

```diff
@@ -29,6 +30,11 @@ export interface AutocompleteOptionProps {
 
   /** the tooltip to display on the Label when the option has been selected */
   tooltip?: string | (() => string);
+  /** the group to display the option in */
+  group?: string;
+
+  /** a unique identifier for the option */
+  uniqueId?: string;
```

Comment:
This is sort of tough since most of the database uses id: number. Would need some sort of data transformation & I am just not sure where to do it. Makes me rethink needing a separate component for the AutoCompleteGrouped. WDYT?

Reply:
> most of the database uses id: number.
>
That's the source of the problem. If unique string based IDs were used then we could expect no conflicts after merging stakeholders and stakeholders group.

In the current situation we need to resolve the ID conflicts so we need a transformation. String-based unique IDs (as you proposed) are a good choice. However if we want to keep it all numbers we can as well design a mapper function that would separate both types i.e.  `({id}) => isGroup ? id : -d`

> Makes me rethink needing a separate component for the AutoCompleteGrouped.
>
IMHO the linear flow inside component is  the most important. If you feel a separate component will look the best then I'm OK with that.